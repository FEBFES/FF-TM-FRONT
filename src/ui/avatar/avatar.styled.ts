import styled, { RuleSet, css } from 'styled-components';
import { AvatarSizesType } from './avatar';

export type StyledVariants<E extends string> = {
  [key in E]?: RuleSet;
};

const sizes: StyledVariants<AvatarSizesType> = {
  '2xs': css`
    width: 12px;
    height: 12px;
  `,
  'xs': css`
    width: 16px;
    height: 16px;
  `,
  's': css`
    min-width: 24px;
    min-height: 24px;
    max-width: 24px;
    max-height: 24px;
  `,
  'm': css`
    width: 32px;
    height: 32px;
  `,
  'l': css`
    width: 48px;
    height: 48px;
  `,
  'xl': css`
    width: 64px;
    height: 64px;
  `,
  '2xl': css`
    width: 128px;
    height: 128px;
  `,
  'fit': css`
    width: 100%;
    height: 100%;
  `,
};

export const SAvatar = styled.img<{
  size: AvatarSizesType;
  bordered: boolean;
}>`
  ${({ size }) => sizes[size]};
  padding: 0;
  margin: 0;
  border-radius: 100px;
  object-fit: cover;
  background: var(--bg-0);
  text-indent: 100%;
  white-space: nowrap;
  overflow: hidden;
  border: ${({ bordered }) =>
    bordered ? '1px solid var(--bg-border)' : 'none'};
`;
