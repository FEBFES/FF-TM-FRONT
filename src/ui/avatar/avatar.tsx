import React, { DetailedHTMLProps } from 'react';
import i18n from 'i18next';
import { SAvatar } from './avatar.styled';

export type AvatarSizesType =
  | '2xs'
  | 'xs'
  | 's'
  | 'm'
  | 'l'
  | 'xl'
  | '2xl'
  | 'fit';

interface AvatarProps
  extends DetailedHTMLProps<
    React.ImgHTMLAttributes<HTMLImageElement>,
    HTMLImageElement
  > {
  bordered?: boolean;
  size?: AvatarSizesType;
}

export const Avatar: React.FC<AvatarProps> = ({
  size = 's',
  bordered = true,
  ...props
}): JSX.Element => {
  return (
    <SAvatar
      alt={i18n.t('utils.any.avatar')}
      bordered={bordered}
      size={size}
      {...props}
    />
  );
};
