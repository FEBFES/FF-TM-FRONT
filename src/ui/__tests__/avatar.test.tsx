import React from 'react';
import {Avatar} from '../avatar/avatar';
import renderer from 'react-test-renderer';

it('SNAPSHOTS: UI - Avatar component - render', () => {
    const elem = renderer.create(<Avatar />).toJSON();
    expect(elem).toMatchSnapshot();
});