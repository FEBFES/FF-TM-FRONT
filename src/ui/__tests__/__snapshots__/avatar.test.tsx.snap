// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOTS: UI - Avatar component - render 1`] = `
<img
  bordered={true}
  className="sc-bcPKhP djjAep"
  size="s"
/>
`;
